Dockerfile

```diff
@@ -38,30 +37,48 @@ RUN --mount=type=cache,target=/go/pkg/mod \
 # Copy the go source
 COPY ./ ./
 
+# Build
+RUN --mount=type=cache,target=/go/pkg/mod \
+    --mount=type=cache,target=/root/.cache/go-build \
+    CGO_ENABLED=0 GOOS=linux GOARCH=${ARCH} go build -ldflags="${LDFLAGS}" -gcflags="${GCFLAGS}" -o manager main.go
```

Comment:
But this is rather considered as a general issue with the build, not PR related, right? If so I would address this as a follow-up to not blowup the PR. Also it will be more visible.

Reply:
You removed a couple of ARGS that I believe break the image.

Reply:
I haven't removed them, I just moved them to the top. AFAIK the ARGs are for the complete Dockerfile, not only for the certain stages. If I was right I will revert it.

Reply:
I'm not saying you are wrong, I'm just asking for a verification to ensure we don't break the docker image. If you believe this should be done in a next PR, I'm fine with that.

Reply:
I can move the necessary ARGs back to the usage to make this more clear. The verification of the images can be done as a follow-up then. WDYT?

Reply:
It is also stated in the documentation that it is only for the current scope

> An `ARG` instruction goes out of scope at the end of the build stage where it was defined. To use an argument in multiple stages, each stage must include the `ARG` instruction.

https://docs.docker.com/reference/dockerfile/#scope